### PR TITLE
cmd/use cmd/install: add --latest to directly install latest release tag

### PR DESCRIPTION
**Pull request by irkode**
implements: #85

-  switch name:  `--latest` switch
- kept `useLatest as name for the flag variable
  - just `if latest` reads bad for me because it does not check _if_ its latest but tells to _use the latest_ version

### CODE REVIEW

File: cmd/install.go [hunk 1]
```diff
@@ -39,28 +39,39 @@ To use this version when version management is disabled in the current
...
+		cobra.CheckErr(err)
+		
+		err = install(useLatest)
 		cobra.CheckErr(err)
 	},
 }
 
 func init() {
 	rootCmd.AddCommand(installCmd)
+	installCmd.Flags().Bool("latest", false, "Install the latest release")
```
Comment (jmooring) [inline cmd/install.go:52]:
```suggestion
	installCmd.Flags().Bool("latest", false, "Install the latest version")
```

File: cmd/use.go [hunk 2]
```diff
@@ -50,15 +50,19 @@ tag to an .hvm file.`,
...
+		err = use(useVersionInDotFile, useLatest)
 		cobra.CheckErr(err)
+
 	},
 }
 
 func init() {
 	rootCmd.AddCommand(useCmd)
 	useCmd.Flags().Bool("useVersionInDotFile", false, "Use the version specified by the "+App.DotFileName+" file\nin the current directory")
+	useCmd.Flags().Bool("latest", false, "Use the latest release")
```
Comment (jmooring) [inline cmd/use.go:65]:
```suggestion
	useCmd.Flags().Bool("latest", false, "Use the latest version")
```

File: cmd/use.go [hunk 3]
```diff
@@ -244,6 +253,16 @@ func (r *repository) fetchTags() error {
 	return nil
 }
 
+// latestTag returns the most recent tag from repository.
```
Comment (jmooring) [inline cmd/use.go:256]:
```suggestion
// getLatestTag returns the most recent tag from repository.
```

**Review by jmooring — CHANGES_REQUESTED** [review]
In addition to these minor comments, this doesn't work when sortAscending = true.

**Comment by jmooring** [comment]
The easiest way to handle this is to add a `latestTag` field to the `repository` struct and populate it from `fetchTags()`.

**Comment by jmooring** [comment]
I guess I wasn't clear. Store the latest version (a string) in the struct.

**Comment by irkode** [comment]
you are always too fast for me - thought we can skip processing the all the tasks  - skip futher explanation.
rethinking - having the latest version available may come handy ...

will update , sry

**Comment by irkode** [comment]
and you are definitely right: much cleaner implementation - sry for the inconvenience

File: cmd/use.go [hunk 4]
```diff
@@ -51,22 +51,27 @@ tag to an .hvm file.`,
...
 
 // A repository is a GitHub repository.
 type repository struct {
-	owner  string         // account owner of the GitHub repository
-	name   string         // name of the GitHub repository without the .git extension
-	tags   []string       // repository tags in semver ascending order
-	client *github.Client // a GitHub API client
+	owner     string         // account owner of the GitHub repository
+	name      string         // name of the GitHub repository without the .git extension
+	tags      []string       // repository tags in semver ascending order
```
Comment (jmooring) [inline cmd/use.go:72]:
```suggestion
	tags      []string       // repository tags
```

File: cmd/use.go [hunk 5]
```diff
@@ -51,22 +51,27 @@ tag to an .hvm file.`,
...
+	latestTag string         // only retrieve the latest tag
```
Comment (jmooring) [inline cmd/use.go:73]:
```suggestion
	latestTag string         // latest repository tag
```

**Review by jmooring — CHANGES_REQUESTED** [review]

**Comment by jmooring** [comment]
OK, this looks good. Thanks.

**Review by jmooring — APPROVED** [review]

**Pull request merged**